src/main/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceService.scala

```diff
@@ -61,4 +71,83 @@ class WorkspaceService(protected val argUserInfo: UserInfo, val rawlsDAO: RawlsD
     }
   }
 
+  def exportWorkspaceAttributes(workspaceNamespace: String, workspaceName: String, filename: String): Future[PerRequestMessage] = {
+    Try(rawlsDAO.getWorkspace(workspaceNamespace, workspaceName)) match {
+      case Failure(regret) => Future(RequestCompleteWithErrorReport(StatusCodes.BadRequest, regret.getMessage))
```

Comment:
2 comments: 1) you don't need a full fledged Future for something we already know the value of. In this case Future.successful creates something that looks like a Future but completes immediately. 2) Why is this a bad request? Seems like you are masking the underlying error. Maybe it was a 404? Maybe it was a communications error?